kubectl-minio/cmd/delete.go

```diff
@@ -100,25 +102,29 @@ func (o *deleteCmd) run() error {
 	// Load Resources
 	emfs, decode := resources.GetFSAndDecoder()
 	crdObj := resources.LoadTenantCRD(emfs, decode)
-	if err := client.CoreV1().ServiceAccounts(o.operatorOpts.Namespace).Delete(context.Background(), helpers.DefaultServiceAccount, v1.DeleteOptions{}); err != nil {
+	if err := client.CoreV1().ServiceAccounts(o.operatorOpts.Namespace).Delete(ctx, helpers.DefaultServiceAccount, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
-	if err := client.AppsV1().Deployments(o.operatorOpts.Namespace).Delete(context.Background(), helpers.DeploymentName, v1.DeleteOptions{}); err != nil {
+	if err := client.AppsV1().Deployments(o.operatorOpts.Namespace).Delete(ctx, helpers.DeploymentName, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
-	if err := extclient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.Background(), crdObj.Name, v1.DeleteOptions{}); err != nil {
+	if err := client.CoreV1().Services(o.operatorOpts.Namespace).Delete(ctx, helpers.DefaultOperatorServiceName, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if err := extclient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, crdObj.Name, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
 	crObj := resources.LoadClusterRole(emfs, decode)
-	if err := client.RbacV1().ClusterRoles().Delete(context.Background(), crObj.Name, v1.DeleteOptions{}); err != nil {
+	if err := client.RbacV1().ClusterRoles().Delete(ctx, crObj.Name, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
-	if err := client.RbacV1().ClusterRoleBindings().Delete(context.Background(), helpers.ClusterRoleBindingName, v1.DeleteOptions{}); err != nil {
+	if err := client.RbacV1().ClusterRoleBindings().Delete(ctx, helpers.ClusterRoleBindingName, v1.DeleteOptions{}); err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
 	consoleResources := resources.LoadConsoleUI(emfs, decode, &o.operatorOpts)
-	if err := deleteConsoleResources(o.operatorOpts, extclient, dynclient, consoleResources); err != nil {
-		return err
+	if err := deleteConsoleResources(o.operatorOpts, extclient, dynclient, consoleResources); err != nil && !k8serrors.IsNotFound(err) {
+		fmt.Println(err)
```

Comment:
we can use `klog` ?

Reply:
this is how most of the things on the plugin is using but yes, makes sense I will add that then.

Reply:
let's change it while we have the time?

Reply:
added klog for info and more detailed warnings.